src/codegen/general.rs

```diff
@@ -177,6 +177,40 @@ pub fn define_shared_type(
     Ok(())
 }
 
+pub fn cfg_deprecated(
+    w: &mut Write,
+    env: &Env,
+    deprecated: Option<Version>,
+    commented: bool,
+    indent: usize,
+) -> Result<()> {
+    if let Some(s) = cfg_deprecated_string(deprecated, env, commented, indent) {
+        try!(writeln!(w, "{}", s));
+    }
+    Ok(())
+}
+
+pub fn cfg_deprecated_string(
+    deprecated: Option<Version>,
+    env: &Env,
+    commented: bool,
+    indent: usize,
+) -> Option<String> {
+    if env.is_totally_deprecated(deprecated) {
```

Comment:
This functions always return false in gdk, due `options.deprecate_by_min_version=false`.
IMHO better add to `Env` function `is_too_low_version`, and write `#[deprecated]` in this case